core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -115,14 +146,25 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     Consumer<ErrorResponse> errorHandler;
     List<UpdateRequirement> requirements;
     List<MetadataUpdate> updates;
+
+    TableMetadata metadataToCommit = metadata;
+    if (encryption() instanceof StandardEncryptionManager) {
```

Comment:
is it possible to add this update to the metadata before we call `commit` ? 
per API contract we should try commit `metadata`
https://github.com/apache/iceberg/blob/aa14aae0111d757f8cded70b87c1d58da3fe272c/core/src/main/java/org/apache/iceberg/TableOperations.java#L45-L63

Reply:
A couple of points,
1. The stated requirement is to "check that the base metadata is current to avoid overwriting updates". This is met. We add a missing field to the new metadata after the check.
2. There are lots of calls to `commit` from various classes. Moving this outside will be technically hard (if it works at all).